tests/test_git_repo.py

```diff
@@ -0,0 +1,88 @@
+import git
+import pytest
+from pytest import Config
+
+from staging.git_repo import get_commit_range_between_refs
+
+
+@pytest.mark.parametrize(
+    "child_ref, ancestor_ref, expected_commit_range",
+    [
+        (
+            "1bcd993b0a8782e1756725127c0cbb6d8c88845e",
+            "1bcd993b0a8782e1756725127c0cbb6d8c88845e",
+            {},
+        ),
+        # 04d1acf7246e8ba11a2c454dc200ce58b53cb807 is the child of 1bcd993b0a8782e1756725127c0cbb6d8c88845e
+        (
+            "1bcd993b0a8782e1756725127c0cbb6d8c88845e",
+            "04d1acf7246e8ba11a2c454dc200ce58b53cb807",
+            {"1bcd993b0a8782e1756725127c0cbb6d8c88845e"},
+        ),
+        # above case, wrong way around: child is the ancestor
+        (
+            "04d1acf7246e8ba11a2c454dc200ce58b53cb807",
+            "1bcd993b0a8782e1756725127c0cbb6d8c88845e",
+            None,
+        ),
+        # The following case breaks a naive git rev-list:
+        #
+        # * |   19c42351 Merge pull request #2442 from HVSharma12/pulseaudio-entrypoint
+        # |\ \
+        # | * | 43074689 Add entrypoint script for PulseAudio container
+        # * | |   e57be2ad Merge pull request #2444 from SUSE/poetry_lock_update
+        # |\ \ \
+        # | * | | 9ed4c2d9 Update poetry.lock
+        # * | | |   9ede9a87 Merge pull request #2439 from SUSE/drop_gcc12
+        # |\ \ \ \
+        # | |/ / /
+        # |/| | |
+        # | * | | fe656acb Remove gcc 12 container
+        # | |/ /
+        # * | |   86990f63 Merge pull request #2441 from SUSE/cosign
+        # |\ \ \
+        # | |/ /
+        # |/| |
+        # | * | 1aa8fc9f update cosign versions
+        # |/ /
+        # * |   37ebaf51 Merge pull request #2435 from HVSharma12/firefox-audio-fix
+        #
+        #
+        # $ git rev-list e57be2ad ^86990f63
+        # e57be2ad998a6b456a405ad967c9707bd350d0e0
+        # 9ede9a8745d557ad918beda987df8e5acd18e9c2
+        # 9ed4c2d9f4e4a09f4f955d3e18ba3c5bc8d525c8
+        # fe656acb9863a900f47257f9f9a240c399854477
+        #
+        # but fe656acb9863a900f47257f9f9a240c399854477 is not a child of 86990f63…
```

Comment:
but it was merged as part of `9ede9a` for the first time, so imho it should be included?

Reply:
forgot to git push :facepalm: